gudrmsglgl/RxCleanNote
# PR: Refactor DataLayer

TestCode & Business Logic Refactor 
- Modify Search Flowable -> Single
- Refactor Business code &  TestCode
- TestCode
   - Separate NoteDataRepositoryTest into NoteDataRepoSearchNoteTest , NoteDataRepoUpdateNoteTest, etc..
   - Add Container (Stub, Verify)
   - Add Case By 4 Step
      1. TestCase Call DataStore Func
      2. TestCase VerifyOrdering DataStore Func
      3. TestCase AssertComplete
      4. TestCase AssertValue
